app/src/main/java/com/github/houseorganizer/houseorganizer/shop/FirestoreShopList.java

```diff
@@ -0,0 +1,154 @@
+    }
+
+    public FirestoreShopList(DocumentReference household, DocumentReference onlineReference){
+        this.household = household;
+        this.onlineReference = onlineReference;
+    }
+
+    public FirestoreShopList(DocumentReference household, DocumentReference onlineReference, List<ShopItem> items){
+        super(items);
+        this.household = household;
+        this.onlineReference = onlineReference;
+    }
+
+    public DocumentReference getHousehold() {
+        return household;
+    }
+
+    @Override
+    public void removeItem(ShopItem shopItem){
+        super.removeItem(shopItem);
+        updateItems();
+    }
+
+    @Override
+    public void removeItem(int index){
+        super.removeItem(index);
+        updateItems();
+    }
+
+    @Override
+    public void addItem(ShopItem item){
+        super.addItem(item);
+        updateItems();
+    }
+
+    @Override
+    public void removePickedUpItems(){
+        super.removePickedUpItems();
+        updateItems();
+    }
+
+    public void setOnlineReference(DocumentReference docRef){
+        onlineReference = docRef;
+    }
+
+    public void setHousehold(DocumentReference household) {
+        this.household = household;
+    }
+
+    public DocumentReference getOnlineReference() {
+        return onlineReference;
+    }
+
+    public Task<Void> updateItems(){
+        if(household == null || onlineReference == null){
+            return Tasks.forCanceled();
+        }
+        List<Map<String, Object>> items = convertItemsListToFirebase(this.getItems());
+
+        return onlineReference.update("items", items);
+    }
+
+    public Task<DocumentSnapshot> refreshItems(){
+        if(onlineReference == null){
+            return Tasks.forCanceled();
+        }
+        return onlineReference.get().continueWith( r -> {
+            DocumentSnapshot snap = r.getResult();
+            setItems(convertFirebaseListToItems((List<Map<String, Object>>) snap.get("items")));
+            return snap;
+        });
+    }
+
+
+
+    private static List<Map<String, Object>> convertItemsListToFirebase(List<ShopItem> shopItemList){
+        List<Map<String, Object>> items = new LinkedList<>();
+
+        for(int i = 0; i < shopItemList.size(); ++i){
+            ShopItem item = shopItemList.get(i);
+            Map<String, Object> itemMap = new HashMap<>();
+            itemMap.put("name", item.getName());
+            itemMap.put("quantity", item.getQuantity());
+            itemMap.put("unit", item.getUnit());
+            itemMap.put("pickedUp", item.isPickedUp());
+            items.add(itemMap);
+        }
+        return items;
+    }
+
+    private static List<ShopItem> convertFirebaseListToItems(List<Map<String, Object>> list){
+        List<ShopItem> items = new LinkedList<>();
+        for(Map<String, Object> m : list){
+            items.add(new ShopItem((String)m.get("name"),
+                    new Long((long) m.get("quantity")).intValue(),
```

Comment:
Here Cirrus suggests to use `Long.valueOf((long) m.get("quantity"))` instead

Reply:
This will give a long instead of a Long which implies I would have to recast it in order to have the IntegerValue ^^'